### PR TITLE
Myscale/improve string pattern match

### PR DESCRIPTION
**Pull request by mpskex**
- Description: Improved prompts for comparator `LIKE` and `CONTAIN`. They are ambiguous under some circumstances like "Answer this questions using papers published by Geoffrey Hinton". In that example LLM tends to use `LIKE` instead of `CONTAIN` to column `authors`, which is actually a column with type list of strings.

We revised the prompt to address this problem:
1. Changed prompt for comparator `LIKE` to `string_pattern_like`
2. Changed prompt for comparator `CONTAIN` to `list_contain`

@hwchase17

### CODE REVIEW

File: langchain/chains/query_constructor/ir.py [hunk 1]
```diff
@@ -75,8 +75,8 @@ class Comparator(str, Enum):
     GTE = "gte"
     LT = "lt"
     LTE = "lte"
-    CONTAIN = "contain"
-    LIKE = "like"
+    CONTAIN = "list_contain"
+    LIKE = "string_pattern_like"
```
Comment (dev2049) [inline langchain/chains/query_constructor/ir.py:79]: this name still feels ambiguous to me, what is the LIKE comparison doing again? is it a regex match?

Comment (mpskex) [reply langchain/chains/query_constructor/ir.py:79]: Personally I think this function should be bounded with datatypes. Unfortunately I didn't find any place to insert the allowed datatypes with those functions. Do you have better idea?

We also explored some new ways to implement [self-query retriever](https://docs.myscale.com/en/sample-applications/openai-function-call/) with openai's function call. We found pretrained LLMs are good SQL translator so it  is probably a good new to SQL based vector databases.

Comment (mpskex) [reply langchain/chains/query_constructor/ir.py:79]: > 

Sorry I probably misunderstand the issue. Do you mean the `LIKE` is ambiguous? Perhaps `SUBSTRING_MATCH` would be a better fit?

**Comment by mpskex** [comment]
Won't fix for now and waiting for dynamic operator prompting strategy.

**Pull request closed without merging**